packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/UserImpl.kt

```diff
@@ -50,11 +50,26 @@ public class UserImpl(
                     // No-op
                 }.freeze()
             )
-            return channel.receive()
+            return@use channel.receive()
                 .getOrThrow()
         }
     }
 
+    override suspend fun remove(): User {
+        Channel<Result<Unit>>(1).use { channel ->
+            RealmInterop.realm_app_remove_user(
+                app.nativePointer,
+                nativePointer,
+                channelResultCallback<Unit, Unit>(channel) {
+                    // No-op
+                }.freeze()
+            )
+            return@use channel.receive()
```

Comment:
This return is inside the channel scope. This is needed in case we have to wait if the user needs to be logged out. 

The user object is returned in https://github.com/realm/realm-kotlin/blob/c0d70753c3da729a46ae214b9926b7b694a38071/packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/UserImpl.kt#L70